.github/workflows/release-side-channel.yml

```diff
@@ -35,7 +43,79 @@ jobs:
           echo "SIDE_CHANNEL_BUCKET=optic-side-channel-production" >> $GITHUB_ENV
       - name: Build NPM packages for S3
         env:
-          DRY_RUN: "false"
+          FLAGS_FILE: ${{ github.event.inputs.flags_file }}
+        run: task release:side-channel:npm_packages
+      - uses: actions/upload-artifact@e448a9b857ee2131e752b06002bf0e093c65e571 # v2.2.2
+        with:
+          name: side-channel-announce
+          path: /tmp/side-channel-outputs/announce.txt
+          if-no-files-found: error
+
+  rust_binaries:
+    strategy:
+      matrix:
+        platform:
+          - os: ubuntu-latest
+            task: diff-engine:build-windows diff-engine:build-linux
+          - os: macos-latest
+            task: diff-engine:build-macos-x86-64
+          - os: macos-11.0
+            task: diff-engine:build-macos-aarch64
+    runs-on: ${{ matrix.platform.os }}
+    steps:
+      - uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f # v2.3.4
+      - name: Install Task
+        uses: Arduino/actions/setup-taskfile@9d04a51fc17daddb0eb127933aaa950af1e3ff97 # no tag
+        with:
+          version: 3.x
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@16499b5e05bf2e26879000db0c1d13f7e13fa3af # v1.0.7
+        with:
+          toolchain: stable
+          profile: minimal
+          override: true
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@32d908adfb55576ba0c59f3c557058e80b5194c3 # https://github.com/aws-actions/configure-aws-credentials/releases/tag/v1.5.3
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+      - name: Set env
+        run: |
+          echo "VERSION_TAG=b${GITHUB_RUN_NUMBER}" >> $GITHUB_ENV
+      - name: Build
+        env:
+          BUILD_ARGS: --all-features --workspace --release
+        run: |
+          task \
+            diff-engine:add-targets \
+            ${{ matrix.platform.task }} \
+            flush-to-disk
+      - run: task diff-engine:locate-binaries
+      - name: Publish
+        env:
+          BUCKET: optic-side-channel-production
+        run: task release:side-channel:diff-engine
+
+  announce:
+    runs-on: ubuntu-latest
+    needs:
+      - npm_packages
+      - rust_binaries
```

Comment:
gha's macos-11.0 availability is still really bad. if we remove `needs:rust_binaries` we could report completion significantly faster most of the time. each diff-engine build matrix uploads its own builds, so by the time npm_packages is done, diff-engine binaries will already be uploaded with the exception of macos-aarch64.

Reply:
haha, pretty sure i should make this change: https://github.com/opticdev/optic/actions/runs/597016999

mac0s-11.0 job was queued for an hour and then randomly failed before starting :\

Reply:
for now, arm builds not being available for an hour or so after everything else will be a thing to keep in mind 🙃